pkg/util/pki/asn1_util.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2023 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pki
+
+import (
+	"encoding/asn1"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"unicode"
+	"unicode/utf8"
+)
+
+// ParseObjectIdentifier parses an object identifier from its string representation.
+func ParseObjectIdentifier(oidString string) (oid asn1.ObjectIdentifier, err error) {
+	if len(oidString) == 0 {
+		return nil, errors.New("zero length OBJECT IDENTIFIER")
+	}
+
+	parts := strings.Split(oidString, ".")
+
+	oid = make(asn1.ObjectIdentifier, 0, len(parts))
+	for _, part := range parts {
+		value, err := strconv.Atoi(part)
+		if err != nil {
+			return nil, err
+		}
+
+		oid = append(oid, value)
+	}
+
+	return oid, nil
+}
+
+type UniversalValue struct {
+	Bytes           []byte
+	Ia5String       string
+	Utf8String      string
```

Comment:
suggestion: as above, following the style guide:

```suggestion
	IA5String       string
	UTF8String      string
```